### ISSUE TITLE
MaxHeight and auto resize on TextBox

### ISSUE THREAD

**Issue by agnauck** [AvaloniaUI/Avalonia#3730]
When having a Textbox with a MaxHeight set on the latest CI build it does not resize automatically as new lines are added. When manual resizing the Windows it seems to get the resize events and adjusts. Same code worked fine on older builds, eg. 0.9.6

```
    <TextBox
        IsReadOnly="False"
        Watermark="Foo"
        FontSize="14"        
        AcceptsReturn="True"
        TextWrapping="Wrap"
        MaxHeight="100"/>
```

See:

![avalonia](https://user-images.githubusercontent.com/1344950/78278037-bfbfc300-7515-11ea-8143-69b37db02dfa.gif)

**Comment by rstm-sf** [comment]
This fix by #3879

**Comment by Gillibald** [comment]
Thanks for testing
